### PR TITLE
[CORE] Unify the logic of switching left/right keys for SHJ

### PR DESCRIPTION
**Pull request by waitinfuture**
## What changes were proposed in this pull request?

This PR unifies the logic in `ShuffledHashJoinExecTransformer#exchangeTable` and
`ShuffledHashJoinExecTransformer#substraitJoinType`, because these two methods
follows the same switching table logic.Unifying these two methods makes the code less error prone.

Also renames `exchangeTable` to `needSwitchChildren`.

## How was this patch tested?
Passes GA.

### CODE REVIEW

File: backends-velox/src/main/scala/io/glutenproject/execution/ShuffledHashJoinExecTransformer.scala [hunk 1]
```diff
@@ -87,7 +87,7 @@ case class ShuffledHashJoinExecTransformer(
    * Returns whether the build and stream table should be exchanged with consideration of build
    * type, planned build side and the preferred build side.
    */
-  override lazy val exchangeTable: Boolean = hashJoinType match {
+  override lazy val switchTable: Boolean = hashJoinType match {
```
Comment (Yohahaha) [inline backends-velox/src/main/scala/io/glutenproject/execution/ShuffledHashJoinExecTransformer.scala:90]: nit: `switchTable` -> `needSwitchTable`.

And switchTable confused me, maybe switchChild make more sense for me, join children does not always be table.

Comment (waitinfuture) [reply backends-velox/src/main/scala/io/glutenproject/execution/ShuffledHashJoinExecTransformer.scala]: Updated, thanks.

**Comment by waitinfuture** [comment]
cc @rui-mo could you please take a look at this PR?

**Review by rui-mo — APPROVED** [review]

**Review by PHILO-HE — APPROVED** [review]
LGTM! Thanks!
cc @zzcclp  please note this change.

**Pull request merged**